src/md/contributing.md

```diff
@@ -58,6 +55,26 @@ To make it easier for the users to set up their development environment, setting
  - [Eclipse](https://github.com/apache/apex-core/tree/master/misc/ide-templates/eclipse)
  - [NetBeans](https://github.com/apache/apex-core/tree/master/misc/ide-templates/netbeans)
 
+## Before Coding
+
+Before starting work, have a JIRA assigned to yourself. If you want to work on a ticket that is assigned to someone else,
+send a courtesy e-mail to the assignee to check if you can take it over. 
+Confirm type, priority and other JIRA fields (often default values are not the best fit).
+
+Before implementing non-trivial changes that result in potentially large code modifications:
+
+* Engage on the mailing list to ensure your time will be well spent, the proposal is welcome by the community
+and does not overlap or conflict with other initiatives.
+* Discuss the design aspects, the larger picture. This may involve discussion, questions,
+suggestions, consensus building towards agreed approach. Or possibly some prototyping. 
+* Capture the outcome or final approach on your JIRA ticket, this will help others to see important
+information without getting into details (including users that may lookup a JIRA from release notes).
+* Certain licenses cannot be used in ASF projects. Ensure contributions don't introduce such dependencies. See
+1. https://www.apache.org/legal/resolved.html#category-x
+1. https://www.apache.org/legal/
+1. https://www.apache.org/legal/resolved.html
+1. https://issues.apache.org/jira/browse/LEGAL
+
 ## Opening Pull Requests
```

Comment:
can you add
- in item 4 that it is possible to assign both Reviewers and Assignees using github (and github provides a suggestion based on the blame)
- in item 5 that both Travis and Jenkins should pass. Should we also mention unit test failures in sub-items?